library/core/src/result.rs

```diff
@@ -1281,16 +1281,26 @@ impl<T, E> Result<T, E> {
     ///
     /// # Examples
     ///
-    /// Basic usage:
+    /// ```
+    /// fn squared_string(x: u32) -> Result<String, &'static str> {
+    ///     Ok((x * x).to_string())
+    /// }
     ///
+    /// assert_eq!(Ok(2).and_then(squared_string), Ok(4.to_string()));
+    /// assert_eq!(Err("not a number").and_then(squared_string), Err("not a number"));
     /// ```
-    /// fn sq(x: u32) -> Result<u32, u32> { Ok(x * x) }
-    /// fn err(x: u32) -> Result<u32, u32> { Err(x) }
     ///
-    /// assert_eq!(Ok(2).and_then(sq).and_then(sq), Ok(16));
-    /// assert_eq!(Ok(2).and_then(sq).and_then(err), Err(4));
-    /// assert_eq!(Ok(2).and_then(err).and_then(sq), Err(2));
-    /// assert_eq!(Err(3).and_then(sq).and_then(sq), Err(3));
+    /// Often used to chain fallible operations that may return [`Err`].
+    ///
+    /// ```
+    /// use std::{io::ErrorKind, path::Path};
+    ///
+    /// let root_modified_time = Path::new("/").metadata().and_then(|md| md.modified());
```

Comment:
pondering: do you know whether this will work on windows?

Reply:
I will test it and get back to you in a bit.

Reply:
Just tested and yes. On windows `/` maps to `C:\` so everything is fine:
```rust
use same_file::is_same_file;
assert!(matches!(is_same_file("/", "C:\\"), Ok(true)));
```
I will add a comment for completeness' sake.